### PR TITLE
add setup_Xpar and related functions

### PR DESCRIPTION
**Pull request by smitdave**
+ setup_Xpar is defined generically and  for each model in exDE
+ make_Xpar_Xname is defined for each model in exDE, with default values
+ make_Xpar_Xinits is defined for each model in exDE (except the trace model), with default values

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -85,3 +85,18 @@ diag_inverse <- function(x) {
 approx_equal <- function(a, b, tol = sqrt(.Machine$double.eps)) {
   abs(a - b) < tol
 }
+
+#' @title Check the length of an input value
+#' @param x a [numeric] object
+#' @param lng a [numeric] object
+#' @param type a [character] string specifying required typeof
+#' @param fixit a [logical] value, if TRUE force length to lng
+#' @return a [logical] value
```
Comment (slwu89) [inline R/utils.R:94]: It should return the same type as the input `x`, not a logical value, right?

**Review by slwu89 — COMMENTED** [review]
This all looks great! It's a lot cleaner than I imagined it.

File: R/utils.R (hunk 1 above)
Comment (smitdave) [reply R/utils.R:94]: yes.

Comment (smitdave) [reply R/utils.R:94]: I'm going to merge and fix this in the next pull request.

**Pull request merged**